frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx

```diff
@@ -311,20 +318,41 @@ const CustomNodeTable: React.FC<CustomNodeTableProps> = ({
 
     const ocsObj = _.cloneDeep(ocsRequestData);
     ocsObj.spec.storageDeviceSets[0].dataPVCTemplate.spec.storageClassName = storageClass;
-    promises.push(k8sCreate(OCSServiceModel, ocsObj));
-
-    Promise.all(promises)
-      .then(() => {
-        history.push(
-          `/k8s/ns/${ocsProps.namespace}/clusterserviceversions/${
-            ocsProps.clusterServiceVersion.metadata.name
-          }/${referenceForModel(OCSServiceModel)}/${ocsObj.metadata.name}`,
-        );
-      })
-      .catch((err) => {
-        setProgress(false);
-        setError(err.message);
-      });
+
+    if (promises) {
```

Comment:
@parthdhanjal 

```
Promise.all(promises)
      .then(() => {
        return k8sCreate(OCSServiceModel, ocsObj);
      })
      .then(() => {
        history.push(
          `/k8s/ns/${ocsProps.namespace}/clusterserviceversions/${
            ocsProps.clusterServiceVersion.metadata.name
          }/${referenceForModel(OCSServiceModel)}/${ocsObj.metadata.name}`,
        );
      })
      .catch((err) => {
        setProgress(false);
        setError(err.message);
      });
```
No need of if else and no need to write separate catch for both scenarios. It can be done in one like I have mentioned above.